src/test/java/com/thealgorithms/datastructures/trees/SplayTreeTest.java

```diff
@@ -0,0 +1,125 @@
+package com.thealgorithms.datastructures.trees;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class SplayTreeTest {
+    @ParameterizedTest
+    @MethodSource("traversalOrders")
+    public void testTraversal(SplayTree.TraverseOrder traverseOrder) {
+        SplayTree tree = createComplexTree();
+        List<Integer> expected = getExpectedTraversalResult(traverseOrder);
+        List<Integer> result = tree.traverse(traverseOrder);
+
+        assertEquals(expected, result);
+    }
+
+    @ParameterizedTest
+    @MethodSource("valuesToTest")
+    public void testSearch(int value) {
+        SplayTree tree = createComplexTree();
+        assertTrue(tree.search(value));
+    }
+
+    @ParameterizedTest
+    @MethodSource("valuesToTest")
+    public void testDelete(int value) {
+        SplayTree tree = createComplexTree();
+
+        assertTrue(tree.search(value));
+        tree.delete(value);
+        assertFalse(tree.search(value));
+    }
+
+    @ParameterizedTest
+    @MethodSource("nonExistentValues")
+    public void testSearchNonExistent(int value) {
+        SplayTree tree = createComplexTree();
+        assertFalse(tree.search(value));
+    }
+
+    @ParameterizedTest
+    @MethodSource("nonExistentValues")
+    public void testDeleteNonExistent(int value) {
+        SplayTree tree = createComplexTree();
+
+        tree.delete(value);
+        assertFalse(tree.search(value));
+    }
+
+    @ParameterizedTest
+    @MethodSource("valuesToTest")
+    public void testDeleteThrowsExceptionForEmptyTree(int value) {
+        SplayTree tree = new SplayTree();
+        assertThrows(IllegalArgumentException.class, () -> tree.delete(value));
+    }
+
+    @ParameterizedTest
+    @MethodSource("valuesToTest")
+    public void testInsertThrowsExceptionForDuplicateKeys(int value) {
+        SplayTree tree = createComplexTree();
+        assertThrows(IllegalArgumentException.class, () -> tree.insert(value));
+    }
+
+    @Test
+    public void testInvalidTraversalOrderExceptionMessage() {
+        SplayTree tree = createComplexTree();
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> tree.traverse(SplayTree.TraverseOrder.INVALID));
+        assertEquals("Invalid traversal order: INVALID", exception.getMessage());
+    }
```

Comment:
One can skip it. I know that that we will loose the `100%` coverage, but it does not matter in this case.

```suggestion
```

One could consider to skip the `switch` statement and use a _polymorphism_. For example, have an interface `TreeTraversal` with 3 implementations `PreOrder`, `InOrder` and `PostOrder`. The `SplayTree` could use an instance of one of them to perform the desired traversal.

Reply:
Well, I will try the idea.

Reply:
I have created the `TreeTraversal` interface and 3 implementations of this interface: `PreOrderTraversal`, `InOrderTraversal`, and `PostOrderTraversal`. You can see them in the next commits.